longest_common_substring.py

```diff
@@ -0,0 +1,15 @@
+def longest_common_substring(X: str, Y: str) -> int:
+    m, n = len(X), len(Y)
+    table = [[0] * (n+1) for _ in range(m+1)]
```

Comment:
explain the logic

Reply:
Initializing a matrix  and storing the length of the common substring